### PR TITLE
[KYUUBI #5731][K8S] Support to cleanup the spark driver pod with specified cleanup strategy

### PR DESCRIPTION
**Pull request by liaoyt**
# :mag: Description

## Describe Your Solution 🔧

A new feature introduced from #5714 supports kyuubi to clean up spark driver pods automatically, but all pod would be clean up without considering app's terminated state.
This PR make user can chose which pods should be delete by setting up a cleanup strategy.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Test locally.

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -1231,13 +1231,20 @@ object KyuubiConf {
       .checkValue(_ > 0, "must be positive number")
       .createWithDefault(Duration.ofMinutes(5).toMillis)
 
-  val KUBERNETES_SPARK_DELETE_DRIVER_POD_ON_TERMINATION_ENABLED: ConfigEntry[Boolean] =
-    buildConf("kyuubi.kubernetes.spark.deleteDriverPodOnTermination.enabled")
-      .doc("If set to true then Kyuubi server will delete the spark driver pod after " +
-        s"the application terminates for ${KUBERNETES_TERMINATED_APPLICATION_RETAIN_PERIOD.key}.")
+  val KUBERNETES_SPARK_DELETE_DRIVER_POD_ON_TERMINATION: ConfigEntry[String] =
+    buildConf("kyuubi.kubernetes.spark.deleteDriverPodOnTermination")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1235]: let's rename the configuration name to `kyuubi.kubernetes.spark.cleanupTerminatedDriverPod`

**Comment by pan3793** [comment]
@liaoyt thanks for catching this, you may refer the wrong ticket in PR description?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (liaoyt) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1235]: OK

**Comment by liaoyt** [comment]
> @liaoyt thanks for catching this, you may refer the wrong ticket in PR description?

oh yes, should refer to #5714, allready fix.

**Comment by pan3793** [comment]
@liaoyt please regenerate the docs, others LGTM

**Review by turboFei — APPROVED** [review]

**Review by zwangsheng — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.8

**Pull request closed without merging**